src/main/webapp/js/components/csv/csv-config-page.jsx

```diff
@@ -0,0 +1,285 @@
+import TextField from '../../utils/form/text-field';
+
+class CsvConfigPage extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      csvConfig: this.props.csvConfig,
+    };
+  }
+
+
+  onSubmit(csvFields) {
+    this.setState({
+      csvConfig: {
+        ...this.state.csvConfig,
+        csvFields,
+        entity: this.props.entityType,
+      },
+    }, () => {
+      if (!this.state.csvConfig.id) {
+        this.props.createCsvConfig(this.state.csvConfig);
+      } else {
+        this.props.saveCsvConfig(this.state.csvConfig);
+      }
+    });
+  }
+
+  getOptions() {
+    return _.map(this.props.fieldConfig, v => ({
+      label: v.displayName,
+      value: v.id,
+    }));
+  }
+
+  getFields(fieldValue) {
+    let hideFormat = true;
+
+    // eslint-disable-next-line max-len
+    const fieldConfig = this.props.fieldConfig[fieldValue.fieldConfigId];
+    if (fieldConfig && (fieldConfig.fieldType === DATE || fieldConfig.fieldType === DATE_TIME)) {
+      hideFormat = false;
+    }
+
+    return [
+      {
+        name: 'fieldConfigId',
+        fieldType: ENUM,
+        displayName: 'Field name',
+        required: true,
+        options: this.getOptions(),
+      },
+      {
+        name: 'fieldName',
+        fieldType: TEXT,
+        displayName: 'CSV name',
+        required: true,
+      },
+      {
+        name: 'format',
+        fieldType: TEXT,
+        displayName: 'Format',
+        required: true,
+        hidden: hideFormat,
+      },
+      {
+        name: 'defaultValue',
+        fieldType: fieldConfig ? fieldConfig.fieldType : TEXT,
+        displayName: 'Default value',
+        relatedField: fieldConfig ? fieldConfig.relatedField : TEXT,
+        relatedEntity: fieldConfig ? fieldConfig.relatedEntity : TEXT,
+        required: fieldConfig ? fieldConfig.required : TEXT,
+        format: fieldConfig ? fieldConfig.format : TEXT,
+      },
+    ];
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  getRelatedFields(relatedField, relatedEntity) {
+    return [
+      {
+        name: 'relatedField',
+        fieldType: RELATION,
+        displayName: 'Related field',
+        required: true,
+        relatedField,
+        relatedEntity,
+      },
+      {
+        name: 'relatedFieldCsvName',
+        fieldType: TEXT,
+        displayName: 'Related field CSV name',
+        required: true,
+      },
+    ];
+  }
+
+  validate = (values) => {
+    const errors = {};
+    errors.csvFields = [];
+
+    _.forEach(values.csvFields, (csvField, key) => {
+      _.forEach(this.getFields(values.csvFields, key), (config) => {
+        const val = csvField[config.name];
+
+        if (config.required && !config.hidden && (_.isNil(val) || val === '')) {
+          errors.csvFields[key] = { [config.name]: `${config.displayName} is required` };
+        }
+      });
+    });
+
+    return errors;
+  };
+
+  deleteConfig() {
```

Comment:
I think you should remove this and delete button should be in the parent, if new config is created there will be no delete button just Cancel